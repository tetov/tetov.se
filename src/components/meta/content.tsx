import React from "react";
import { Helmet } from "react-helmet";

type Prop = {
  title: string;
  excerpt?: string;
  lang?: string;
  type?: string;
  dateCreated?: string;
};

const MetaContent: React.FC<Prop> = ({
  title,
  excerpt,
  type,
  lang,
  dateCreated,
}) => (
  <Helmet>
    {lang && <html lang={lang} />}

    <title>{title}</title>
    <meta property="og:title" content={title} />

    {excerpt && <meta name="description" content={excerpt} />}
    {excerpt && <meta property="og:description" content={excerpt} />}

    {type && <meta property="og:type" content={type} />}

    {dateCreated && type === "article" && (
      <meta property="article:published_time" content={dateCreated} />
    )}
  </Helmet>
);

export default MetaContent;
