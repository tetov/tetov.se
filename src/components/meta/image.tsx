import querySiteMetadata from "~/hooks/query-site-metadata";
import React from "react";
import { Helmet } from "react-helmet";

type Prop = {
  src?: string;
  alt?: string;
};

const MetaImage: React.FC<Prop> = ({ src, alt }) => {
  const { siteURL } = querySiteMetadata();
  return (
    <Helmet>
      {src && <meta property="og:image" content={siteURL + src} />}
      {src && alt && <meta property="og:image:alt" content={alt} />}
    </Helmet>
  );
};

export default MetaImage;
