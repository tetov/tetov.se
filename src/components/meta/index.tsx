export { default as MetaContent } from "./content";
export { default as MetaImage } from "./image";
export { default as MetaSiteWide } from "./site-wide";
export type { MetaSiteWide as MetaProp } from "./site-wide";
