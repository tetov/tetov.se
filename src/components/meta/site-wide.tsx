import { graphql, useStaticQuery } from "gatsby";
import React from "react";
import { Helmet } from "react-helmet";
import logo from "src/images/logo.png";

export type MetaSiteWide = {
  pathName: string;
  data: GatsbyTypes.MetaSiteWideQuery;
};

export const PureMetaSiteWide: React.FC<{pathName: string, data: GatsbyTypes.MetaSiteWideQuery}> = ({
  pathName,
  data,
}) => {
  const {
    site: {
      siteMetadata: { title, siteURL, lang, description },
    },
    allContactData,
  } = data;
  const twitterUsername = allContactData.nodes[0].username;

  return (
    <Helmet titleTemplate={`%s | ${title}`} defaultTitle={title}>
      <html lang={lang || "en"} prefix="og: http://ogp.me/ns#" />

      <meta property="og:site_name" content={title} />

      <meta property="og:url" content={siteURL + pathName} />

      <meta property="og:type" content="website" />

      <meta property="og:title" content={title} />

      <meta name="description" content={description.trim()} />
      <meta property="og:description" content={description.trim()} />

      <meta property="og:image" content={siteURL + logo} />

      <meta name="twitter:card" content="summary" />
      <meta name="twitter:site" content={twitterUsername} />
      <meta name="twitter:creator" content={twitterUsername} />
    </Helmet>
  );
};

const MetaSiteWide: React.FC<{pathName: string}> = ({ pathName }) => {
  const data = useStaticQuery<GatsbyTypes.MetaSiteWideQuery>(
    graphql`
      query MetaSiteWide {
        site {
          siteMetadata {
            title
            siteURL
            lang
            description
          }
        }

        allContactData(filter: { label: { eq: "twitter" } }, limit: 1) {
          nodes {
            username
          }
        }
      }
    `
  );
  return <PureMetaSiteWide pathName={pathName} data={data} />;
};

export default MetaSiteWide;
