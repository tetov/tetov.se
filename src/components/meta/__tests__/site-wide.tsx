import React from "react";
import renderer from "react-test-renderer";
import { PureMetaSiteWide as MetaSiteWide} from "src/components/meta/site-wide";

describe("Header", () => {
  it("renders correctly", () => {
    const data: GatsbyTypes.MetaSiteWideQuery = {
      site: {
        siteMetadata: {
          title: "tetov's projects & more",
          siteURL: "https://tetov.se",
          lang: "en",
          description: "there no 'more', is there?",
        },
      },
      allContactData: {
        nodes: [
          {
            username: "antontetov",
          },
        ],
      },
    };
    const tree = renderer
      .create(<MetaSiteWide pathName={"Test"} data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
