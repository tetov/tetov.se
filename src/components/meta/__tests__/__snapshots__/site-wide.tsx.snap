// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="mock-helmet"
  defaultTitle="tetov's projects & more"
  titleTemplate="%s | tetov's projects & more"
>
  <html
    lang="en"
    prefix="og: http://ogp.me/ns#"
  />
  <meta
    content="tetov's projects & more"
    property="og:site_name"
  />
  <meta
    content="https://tetov.seTest"
    property="og:url"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="tetov's projects & more"
    property="og:title"
  />
  <meta
    content="there no 'more', is there?"
    name="description"
  />
  <meta
    content="there no 'more', is there?"
    property="og:description"
  />
  <meta
    content="https://tetov.setest-file-stub"
    property="og:image"
  />
  <meta
    content="summary"
    name="twitter:card"
  />
  <meta
    content="antontetov"
    name="twitter:site"
  />
  <meta
    content="antontetov"
    name="twitter:creator"
  />
</div>
`;
